src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonWebToken.PayloadClaimSet.cs

```diff
@@ -78,8 +79,10 @@ internal JsonClaimSet CreatePayloadClaimSet(ReadOnlySpan<byte> byteSpan)
                     }
                     else if (reader.ValueTextEquals(JwtPayloadUtf8Bytes.Iss))
                     {
-                        _iss = JsonSerializerPrimitives.ReadString(ref reader, JwtRegisteredClaimNames.Iss, ClassName, true);
-                        claims[JwtRegisteredClaimNames.Iss] = _iss;
+                        
+                        IssuerUtf8 = JsonSerializerPrimitives.ReadStringUtf8(ref reader, JwtRegisteredClaimNames.Jti, ClassName, true).ToArray();
```

Comment:
Using the bytes in ValueSpan straight up doesn't work if the characters are escaped. Calling reader.GetString actually unescapes the characters if needed.
There is a reader.ValueIsEscaped property but it's only available on .NET 8+.

Reply:
Use https://learn.microsoft.com/en-us/dotnet/api/system.text.json.utf8jsonreader.copystring?view=net-8.0#system-text-json-utf8jsonreader-copystring(system-span((system-byte)))